app/workers/asset_manager_create_whitehall_asset_worker.rb

```diff
@@ -12,15 +12,12 @@ def perform(file_path, legacy_url_path, draft = false, model_class = nil, model_
 
     if model_class && model_id
       model = model_class.constantize.find(model_id)
-      if model.respond_to?(:access_limited?) && model.access_limited?
-        authorised_user_uids = AssetManagerAccessLimitation.for(model)
-        asset_options[:access_limited] = authorised_user_uids
-      end
+      asset_options.merge!(access_options(model))
     end
 
     asset_manager.create_whitehall_asset(asset_options)
 
-    if model
+    if is_attachment
```

Comment:
We can determine `is_attachment` within this worker, rather than relying on the storage class to work it out for us.

Reply:
this now reverted back to the previous way of deriving this in the worker (albeit with some hopefully better naming) so resolving this comment